src/module.cpp

```diff
@@ -24,7 +24,9 @@ struct module_impl
 {
     // A list is used to keep references to an instruction stable
     std::list<instruction> instructions;
+    std::list<module> sub_modules;
```

Comment:
The module should not own the submodules, as these can be referenced across multiple modules. The modules should be owned by the program class.